### PR TITLE
Remove an unresolved extern from the GPU header

### PR DESCRIPTION
**Pull request by e-kayrakli**
We had a device `extern` for `chpl_nodeID` to be able to implement device
versions of the functions in `chpl-gen-includes.h`. But, we don't really have a
Chapel device runtime where something like that could be defined. So, so far, we
have been getting "unresolved extern" warnings from `ptxas` and have been
PREDIFF'ing/mentally ignoring that.

But the problem is that the "chpl-gen" functions we have for device don't even
get called. We need them to be able to compile all the standard library
foralls/foreachs. So, the previous "unresolved extern" was probably just getting
junk value, anyways.

Long story short, this PR removes that extern, replaces its uses with just `0`,
and adds bunch of comments to the header to make the purpose of the header
clearer and the path forward for it.

Test:

- [x] gpu/native/multi* with gasnet
- [x] gpu/native with local

### CODE REVIEW

**Review by stonea — APPROVED** [review]
Great to no longer get that warning!

Out of curiosity, do you know what would cause us to call these functions from within a kernel: (`chpl_task_getRequestedSubloc()`, `chpl_gen_getLocaleID()`, and `get_chpl_nodeID()`).

Presumably, these are returning the wrong values when we're not on Locale[0], so it might be good to come up with a breaking test and create an issue for it.

**Comment by e-kayrakli** [comment]
As far as I remember and can see with quick grepping:

- `chpl_gen_getLocaleID` is a function called by the compiler to generate a locale ID for a wide pointer. With your PR adding an implicit `local` block, this shouldn't happen. (I have a vague memory where `local` block wasn't suppressing all wide pointers or something?). IOW, maybe we can remove it, now that we have the implicit `local` block? Though I don't find that necessary as we need this function to work correctly when called from the kernel, if we want GPU-driven communication to be wide-pointer-based, too (can't think of any other reasonable way myself). 
- `get_chpl_nodeID` is renamed and used as an extern to implement `chpl_nodeID` in the module code. I thought we used this to get the node ID as an int during standard library initialization, but I see it being used in the BigInteger module, too. Maybe it has something to do with C interop. So, maybe an easy way to trigger it is to use some BigInteger stuff in a multilocale GPU setting on a Locale other than 0? Which may not work for other reasons, too.
- `chpl_task_getRequestedSubloc()` I think we use this heavily as part of the GPU effort. There might be some calls to the device version of this in a nested foreach scenario for checking where we are executing the inner foreach, for example. This one's also used by `chpl_gen_getLocaleID`
